### PR TITLE
Prevent indexing when blog is deleted or not public

### PR DESCRIPTION
**Pull request by allan23**
Add additional hook for when an blog is deleted.  Also conditional logic to prevent posts from getting indexed if blog is not indexable.

### CODE REVIEW

**Comment by oscarssanchez** [comment]
Hi @allan23 ,

I got posts to index without any issues and only sites that were in the public category get their posts indexed. 

However, these changes are also breaking creating a new post for me on any site, with this error: 

error

These are my test sites:

error_settings

Individual settings for each site: 

settings_1

settings_2

settings_3

settings_4

Steps to reproduce:

1.- Edit site settings and set them to deleted/archived/non-public
2.- Try to create a post on each individual site from the dashboard for any site.

File: classes/class-ep-sync-manager.php [hunk 1]
```diff
@@ -263,14 +265,35 @@ public function sync_post( $post_id, $blocking = true ) {
...
+	protected function is_site_indexable() {
+		if ( ! is_multisite() ) {
+			return true;
+		}
+		$blog_id = get_current_blog_id();
+		$sites   = ep_get_sites();
+		foreach ( $sites as $site ) {
+			if ( $blog_id === absint( $site['blog_id'] ) ) {
+				return true;
+				break;
```
[the remark this answers is not in the document]
Comment (tott) [reply classes/class-ep-sync-manager.php]: I would also recommend just querying the site types in the arguments of get_sites/wp_get_sites as outlined here https://developer.wordpress.org/reference/functions/get_sites this makes additional logic obsolete

[the remark this answers is not in the document]
Comment (allan23) [reply classes/class-ep-sync-manager.php]: @tott PR updated to simplify the logic.

File: classes/class-ep-api.php [hunk 2]
```diff
@@ -2011,6 +2011,11 @@ public function get_sites( $limit = 0 ) {
 			$sites = array();
 
 			foreach ( $site_objects as $site ) {
+
+				if ( ! $site->public || $site->spam || $site->deleted ) {
```
Comment (tott) [inline classes/class-ep-api.php]: Sounds like this is obsolete now :)

Comment (allan23) [reply classes/class-ep-api.php]: Yep! This has been removed.

**Pull request merged**